lib/utils/note-utils.js

```diff
@@ -1,38 +1,58 @@
 import removeMarkdown from 'remove-markdown';
 import { isFunction } from 'lodash';
 
-/**
- * Matches the title and excerpt in a note's content
- *
- * Both the title and the excerpt are determined as
- * content starting with something that isn't
- * whitespace and leads up to a newline or line end
- *
- * @type {RegExp} matches a title and excerpt in note content
- */
-const noteTitleAndPreviewRegExp = /\s*(\S[^\n]*)\s*(\S[^\n]*)?/;
+const maxTitleChars = 64;
+const maxPreviewChars = 200;
+const maxPreviewLines = 4; // probably need to adjust if we're in the comfy view
+
+const matchUntilLimit = ( pattern, source, preview = '', lines = 0 ) => {
+    const match = pattern.exec( source );
+    // must match, must have no more than four lines, must not be longer than N=10 characters
+    if ( ! match || lines > maxPreviewLines || preview.length > maxPreviewChars ) {
+      return preview;
+    }
+
+    const [ , chunk ] = match;
+    return matchUntilLimit( pattern, source, preview + chunk, lines + 1 );
+  }
 
 /**
- * Matches the title and excerpt in a note's content skipping opening # from title
- *
- * Both the title and the excerpt are determined as
- * content starting with something that isn't
- * whitespace and leads up to a newline or line end
+ * Returns a string with markdown stripped
  *
- * @type {RegExp} matches a title and excerpt in note content skipping opening #'s from title and preview
+ * @param {String} inputString string for which to remove markdown
+ * @returns {String} string with markdown removed
  */
-const noteTitleAndPreviewMdRegExp = /(?:#{0,6}\s+)?(\S[^\n]*)\s*(?:#{0,6}\s+)?(\S[^\n]*)?/;
-
-export const titleCharacterLimit = 200;
+const removeMarkdownWithFix = ( inputString ) => {
+  // Workaround for a bug in `remove-markdown`
+  // See https://github.com/stiang/remove-markdown/issues/35
+  return removeMarkdown(inputString.replace(/(\s)\s+/g, '$1'), {
+    stripListLeaders: false,
+  });
+  ;
+}
 
-// Ample amount of characters for the 'Expanded' list view
-export const previewCharacterLimit = 300;
+const getTitle = ( content, isNoteMarkdown ) => {
+  const titlePattern = new RegExp( `\s*([^\n]{1,${ maxTitleChars }})`, 'g' );
+  const titleMatch = titlePattern.exec( content );
+  if ( ! titleMatch ) {
+    return 'New Note...';
+  }
+  const [ , rawTitle ] = titleMatch;
+  if ( isNoteMarkdown ) {
+    let parsed = removeMarkdownWithFix( rawTitle )
+    return parsed.length && parsed || rawTitle;
+  }
+  return rawTitle;
+}
 
-// Defaults for notes with empty content
-export const defaults = {
-  title: 'New Note...',
-  preview: '',
-};
+const getPreview = ( content, isNoteMarkdown ) => {
+  const previewPattern = new RegExp( `[^\n]*\n+[ \t]*([^]{0,${ maxPreviewChars }})`, 'g' );
+  let preview = matchUntilLimit( previewPattern, content );
+  if ( isNoteMarkdown ) {
+    preview = removeMarkdownWithFix( preview );
+  }
+  return preview;
```

Comment:
by combining we have the chance to eliminate the mutation and clarify the control-flow

```js
const preview = matchUntilLImit( previewPattern, content );

return isNoteMarkdown
  ? removeMarkdownWithFix( preview )
  : preview;
```

Reply:
Wow, I always prefered my version. Never thought about the mutation.

Reply:
mutation scares me and gives me a headache. I like knowing that once I find an assignment in the code I don't have to worry about interpreting the rest of the code in my head to learn if the value might be reassigned outside my purview